### ISSUE TITLE
Allow any files in `_patterns` to be copied to `publicDir`

### ISSUE THREAD

**Issue by EvanLovely** [pattern-lab/patternlab-php-core#45]
Since we've recently gotten in the ability to use folders at the pattern level, I've been keeping more assets related to patterns next to the pattern files and I think it'd be great if those ended up next to the compiled HTML (possibly after turning on a config option as I realize copying lots of large files could slow compiles down). Here's my use case:

I'm using the pattern descriptions markdown file to also show the comp image of that pattern. However, I've got to reference to from the public directory to it's location in the source directory, like so:

![2016-05-26 at 11 05 am](https://cloud.githubusercontent.com/assets/569699/15584986/e548e668-2331-11e6-899d-c95a9ac6687b.png)

What would be great is if an image link could be as simple as:

    ![](comp.png)

Thanks for considering!

**Comment by dmolsen** [comment]
Ok, saw those `source/` links in your earlier example and it made me squeamish. I think this could be done with a plug-in. I'm not keen on making this `core` as then it becomes spec and yet another thing for the Node team to copy. Maybe there could be opt-in file types with some sane defaults auto-populated. It would probably only work with that folder structure. Doable with some caveats at least.

**Comment by EvanLovely** [comment]
Yeah, links like that make me squeamish too; very brittle. Adds a lot of value though! Helps to make conversations with clients less about pages and more about components for sure. If this was in core; here's how I'd see it set up as a config array of file types to copy over with a default of just `twig` or `mustache`. So:

```yml
copyFileTypesToPublic:
  - twig
```

I think this would be helpful as part of the core spec as I see this as very helpful. Especially as more people realize that they can document each pattern with markdown. Having this in the code slide-up is really nice. Perhaps the styleguide kit might also want to make a tab to show off the `scss` file? Your call!

**Comment by dmolsen** [comment]
Plug-ins can do the following:

* add options to the config (_on install_)
* add tab & panel combos to the modals (_this is all the work being done in `feature-panels`_)

So, honestly, the only thing this feature gets from being in `core` is that PL/Node _has_ to copy it. And I'm definitely not picking sides on non-PHP tech (e.g. Sass). I'm willing to write this feature and make it "official." I'm just not willing to bake it into `core`.

**Comment by EvanLovely** [comment]
A plugin would be great!

**Comment by dmolsen** [comment]
In thinking through this more I need to write hooks in Core to make this happen in an efficient manner. It also opens the possibility for redoing/simplifying panels. I need to write up my ideas so they can go up for approval as part of the spec for Pattern Lab. The changes would also affect how PL/Node works.
